crypto/aes/asm/aes-riscv64.pl

```diff
@@ -0,0 +1,1709 @@
+
+    ld      $T6,24($UKEY)
+    srli    $T7,$T6,32
+    sd      $T6,24($KEYP)
+
+    # bits == 256
+    j       3f
+___
+
+$code .= <<___;
+1:
+    addi    $KEYP,$KEYP,16
+1:
+___
+$code .= do_enc_lookup($T4,$T3,[8,16,24,0],[$T4,$T5,$T6,$T7]);
+
+$code .= <<___;
+    add     $T5,$RCON,$loopcntr # rcon[i] (i increments by 4 so it can double as
+                                # a word offset)
+    lwu     $T5,0($T5)
+
+    addi    $loopcntr,$loopcntr,4
+    li      $I0,10*4
+
+    xor     $T0,$T0,$T4
+    xor     $T0,$T0,$T5
+    xor     $T1,$T1,$T0
+    xor     $T2,$T2,$T1
+    xor     $T3,$T3,$T2
+
+    sw      $T0,0($KEYP)
+    sw      $T1,4($KEYP)
+    sw      $T2,8($KEYP)
+    sw      $T3,12($KEYP)
+
+    addi    $KEYP,$KEYP,16
+
+
+    bne     $loopcntr,$I0,1b
+    j       4f
+___
+$code .= <<___;
+2:
+    addi    $KEYP,$KEYP,24
+2:
+___
+$code .= do_enc_lookup($T6,$T5,[8,16,24,0],[$T6,$T7,$T8,$T9]);
+
+$code .= <<___;
+    add     $T7,$RCON,$loopcntr # rcon[i] (i increments by 4 so it can double as
+                                # a word offset)
+    lwu     $T7,0($T7)
+
+    addi    $loopcntr,$loopcntr,4
+    li      $I0,8*4
+
+    xor     $T0,$T0,$T6
+    xor     $T0,$T0,$T7
+    xor     $T1,$T1,$T0
+    xor     $T2,$T2,$T1
+    xor     $T3,$T3,$T2
+
+    sw      $T0,0($KEYP)
+    sw      $T1,4($KEYP)
+    sw      $T2,8($KEYP)
+    sw      $T3,12($KEYP)
+
+    beq     $loopcntr,$I0,4f
+
+    xor     $T4,$T4,$T3
+    xor     $T5,$T5,$T4
+    sw      $T4,16($KEYP)
+    sw      $T5,20($KEYP)
+
+    addi    $KEYP,$KEYP,24
+    j       2b
+___
+$code .= <<___;
+3:
+    addi    $KEYP,$KEYP,32
+3:
+___
+$code .= do_enc_lookup($T8,$T7,[8,16,24,0],[$T8,$T9,$T10,$T11]);
+
+$code .= <<___;
+    add     $T9,$RCON,$loopcntr # rcon[i] (i increments by 4 so it can double as
+                                # a word offset)
+    lwu     $T9,0($T9)
+
+    addi    $loopcntr,$loopcntr,4
+    li      $I0,7*4
+
+    xor     $T0,$T0,$T8
+    xor     $T0,$T0,$T9
+    xor     $T1,$T1,$T0
+    xor     $T2,$T2,$T1
+    xor     $T3,$T3,$T2
+
+    sw      $T0,0($KEYP)
+    sw      $T1,4($KEYP)
+    sw      $T2,8($KEYP)
+    sw      $T3,12($KEYP)
+
+    beq     $loopcntr,$I0,4f
+___
+$code .= do_enc_lookup($T8,$T3,[0,8,16,24],[$T8,$T9,$T10,$T11]);
+$code .= <<___;
+    xor     $T4,$T4,$T8
+    xor     $T5,$T5,$T4
+    xor     $T6,$T6,$T5
+    xor     $T7,$T7,$T6
+    sw      $T4,16($KEYP)
+    sw      $T5,20($KEYP)
+    sw      $T6,24($KEYP)
+    sw      $T7,28($KEYP)
+
+    addi    $KEYP,$KEYP,32
+    j       3b
+
+4:  # return 0
+    li      a0,0
+5:  # return a0
+___
+$code .= load_regs();
+$code .= <<___;
+    ret
+___
+
+clear_regs();
+
+################################################################################
+# Register assignment for AES_set_decrypt_key
+################################################################################
+
+# Function arguments (x10-x12 are a0-a2 in the ABI)
+# Input block pointer, output block pointer, key pointer
```

Comment:
The comment here is incorrect (found this when editing my PR)

Reply:
Thank you for catching this -- stale comments are always frustrating!

Reply:
You fixed it in `Register assignment for AES_set_[en/de]crypt_key` but the comment for `Register assignment for AES_set_decrypt_key` is still stale.

BTW, `Register assignment for AES_set_[en/de]crypt_key` should be `Register assignment for AES_set_encrypt_key` as you have `clear_regs` in line 953. Thus these reg assignments in 670 do not cover `decrypt`

Reply:
Fixed in latest push, thank you!